File: fix_syntax_errors_simple.js
#!/usr/bin/env node

import fs from 'fs';
import path from 'path';
import { execSync } from 'child_process';

console.log('🔧 Starting comprehensive syntax error fixing...');

// Function to fix common syntax errors
function fixSyntaxErrors(content) {
  let fixed = content;

  // Fix unterminated string literals
  fixed = fixed.replace(/'([^']*)$/gm, (match, str) => {
    if (str && !str.includes("'") && !str.includes('"')) {
      return `"${str}"`;
    }
    return match;
  });

  // Fix missing closing tags for fragments
  fixed = fixed.replace(/<>([^<]*(?:<(?!/>)[^<]*)*?)(?=<[^/]|$)/g, (match, inner) => {
    if (!inner.includes('</>')) {
      return match + inner + '</>';
    }
    return match;
  });

  // Fix missing closing div tags
  fixed = fixed.replace(/<div[^>]*>([^<]*(?:<(?!/div>)[^<]*)*?)(?=<[^/]|$)/g, (match, inner) => {
    if (!inner.includes('</div>')) {
      return match + inner + '</div>';
    }
    return match;
  });

  // Fix missing closing braces
  fixed = fixed.replace(/\{([^}]*)$/gm, (match, inner) => {
    if (inner.trim() && !inner.includes('}')) {
      return match + '}';
    }
    return match;
  });

  // Fix missing closing parentheses
  fixed = fixed.replace(/\(([^)]*)$/gm, (match, inner) => {
    if (inner.trim() && !inner.includes(')')) {
      return match + ')';
    }
    return match;
  });

  // Fix malformed JSX attributes
  fixed = fixed.replace(/className="([^"]*)"([^>]*>)/g, 'className="$1"$2');
  fixed = fixed.replace(/className='([^']*)'([^>]*>)/g, 'className="$1"$2');

  return fixed;
}

// Function to fix specific problematic files
function fixSpecificFiles() {
  const problematicFiles = [
    'app/components/Footer.tsx',
    'app/components/Navigation.tsx',
    'app/components/AdvancedSEOOptimizer.tsx',
    'app/components/PerformanceOptimizer.tsx',
    'app/components/lazyUtils.tsx'
  ];

  for (const filePath of problematicFiles) {
    try {
      if (fs.existsSync(filePath)) {
        const content = fs.readFileSync(filePath, 'utf8');
        const fixed = fixSyntaxErrors(content);
        
        if (fixed !== content) {
          fs.writeFileSync(filePath, fixed);
          console.log(`✅ Fixed: ${filePath}`);
        }
      }
    } catch (error) {
      console.error(`❌ Error fixing ${filePath}:`, error.message);
    }
  }
}

// Function to fix page files with missing closing tags
function fixPageFiles() {
  const pageFiles = [
    'app/5g-solutions/page.tsx',
    'app/about/page.tsx',
    'app/ai-customer-sentiment-tracker/page.tsx',
    'app/ai-data-analytics-pro/page.tsx',
    'app/ai-financial-analysis/page.tsx',
    'app/ai-healthcare-diagnostics/page.tsx',
    'app/ai-holographic-workspace/page.tsx',
    'app/ai-hr-recruitment-pro/page.tsx',
    'app/ai-image-recognition-pro/page.tsx',
    'app/ai-powered-devops/page.tsx',
    'app/ai-services/page.tsx',
    'app/ai-solutions/page.tsx',
    'app/blog/page.tsx'
  ];

  for (const filePath of pageFiles) {
    try {
      if (fs.existsSync(filePath)) {
        let content = fs.readFileSync(filePath, 'utf8');
        
        // Fix missing closing tags
        if (content.includes('<>') && !content.includes('</>')) {
          content = content.replace(/(<>[^<]*(?:<(?!/>)[^<]*)*?)(?=<[^/]|$)/g, (match) => {
            if (!match.includes('</>')) {
              return match + '</>';
            }
            return match;
          });
        }

        if (content.includes('<div') && !content.includes('</div>')) {
          content = content.replace(/(<div[^>]*>[^<]*(?:<(?!/div>)[^<]*)*?)(?=<[^/]|$)/g, (match) => {
            if (!match.includes('</div>')) {
              return match + '</div>';
            }
            return match;
          });
        }

        fs.writeFileSync(filePath, content);
        console.log(`✅ Fixed page: ${filePath}`);
      }
    } catch (error) {
      console.error(`❌ Error fixing ${filePath}:`, error.message);
    }
  }
}

// Main execution
function main() {
  console.log('🔧 Fixing specific problematic files...');
  fixSpecificFiles();
  
  console.log('🔧 Fixing page files...');
  fixPageFiles();

  // Run type check to see remaining errors
  console.log('\n🔍 Running type check to verify fixes...');
  try {
    execSync('pnpm run type-check', { stdio: 'inherit' });
    console.log('✅ Type check passed!');
  } catch (error) {
    console.log('⚠️  Some type errors remain, but many have been fixed');
  }
}

main();

File: fix_syntax_errors_simple.py
#!/usr/bin/env python3
"""
Simple script to fix common syntax errors in TypeScript/JSX files
"""

import os
import re
import glob

def fix_syntax_errors(file_path):
    """Fix common syntax errors in a single file"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        
        # Fix 1: Remove trailing comma after function declaration
        content = re.sub(r'const\s+\w+:\s*React\.FC\s*=\s*\(\)\s*=>\s*\{,', 'const \\1: React.FC = () => {', content)
        
        # Fix 2: Fix malformed h1 tags - <h1Ai -> <h1>Ai
        content = re.sub(r'<h1([A-Z][a-zA-Z\s]*)', r'<h1>\1', content)
        
        # Fix 3: Fix missing closing tags in h1
        content = re.sub(r'<h1>([^<]+)\n\s*</h1>', r'<h1>\1</h1>', content)
        
        # Fix 4: Fix malformed JSX with comma issues
        content = re.sub(r'className="([^"]*)"\s*,', r'className="\1"', content)
        
        # Fix 5: Fix missing semicolons after const declarations
        content = re.sub(r'const\s+\w+\s*=\s*new\s+Date\(\)\.getFullYear\(\)\n', 'const \\1 = new Date().getFullYear();\n', content)
        
        # Fix 6: Fix malformed div structure
        content = re.sub(r'</div><div className="([^"]*)"\s*>,', r'</div><div className="\1">', content)
        
        # Fix 7: Fix malformed JSX structure with commas
        content = re.sub(r'<div className="min-h-screen[^"]*">\s*</div><div className="[^"]*">,', 
                        lambda m: m.group(0).replace(',', ''), content)
        
        # Only write if content changed
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"Fixed syntax errors in: {file_path}")
            return True
        
        return False
        
    except Exception as e:
        print(f"Error processing {file_path}: {e}")
        return False

def main():
    """Main function to fix syntax errors in all files"""
    patterns = [
        'app/**/*.tsx',
        'app/**/*.ts',
        'components/**/*.tsx',
        'components/**/*.ts'
    ]
    
    files_processed = 0
    files_fixed = 0
    
    for pattern in patterns:
        for file_path in glob.glob(pattern, recursive=True):
            if os.path.isfile(file_path):
                files_processed += 1
                if fix_syntax_errors(file_path):
                    files_fixed += 1
    
    print(f"\nProcessed {files_processed} files")
    print(f"Fixed syntax errors in {files_fixed} files")

if __name__ == "__main__":
    main()